baseplate/server/__init__.py

```diff
@@ -83,7 +83,10 @@ def make_listener(endpoint):
         sock.listen(128)
         return sock
     else:
-        return socket.fromfd(fileno, socket.AF_INET, socket.SOCK_STREAM)
+        assert fd_count > 0, "Running under Einhorn  but no sockets were bound."
+        fileno = int(os.environ["EINHORN_FD_0"])
```

Comment:
I've added a docstring to this function and a few others in this file to hopefully explain the Einhorn logic. Do these answer your questions?